src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const clone = [];
+  let currentState = { ...state };
+
+  for (const object of actions) {
+    const { type, extraData = {}, keysToRemove = [] } = object;
+
+    if (type === 'addProperties') {
+      currentState = { ...currentState, ...extraData };
+    }
+
+    if (type === 'removeProperties') {
+      currentState = { ...currentState };
 
+      for (const key of keysToRemove) {
+        delete currentState[key];
+      }
+    }
+
+    if (type === 'clear') {
+      currentState = {};
+    }
```

Comment:
[CODE STYLE]: Rather than using multiple `if` statements, a `switch` statement would be more appropriate here since you are checking the value of 'type' against a limited set of conditions. Also, include a `default` case to handle any unexpected 'type' values for error handling.